samples/test/util.py

```diff
@@ -282,35 +276,37 @@ def run_v2_pipeline(
 ):
     import tempfile
     import subprocess
-    original_pipeline_job = tempfile.mktemp(suffix='.json',
-                                            prefix="original_pipeline_job")
+    pipeline_spec_file = tempfile.mktemp(suffix='.json',
+                                         prefix="original_pipeline_job")
     kfp.v2.compiler.Compiler().compile(pipeline_func=fn,
-                                       package_path=original_pipeline_job)
+                                       package_path=pipeline_spec_file)
 
     # remove following overriding logic once we use create_run_from_job_spec to trigger kfp pipeline run
-    with open(original_pipeline_job) as f:
-        pipeline_job_dict = json.load(f)
-    for component in [pipeline_job_dict['pipelineSpec']['root']] + list(
-            pipeline_job_dict['pipelineSpec']['components'].values()):
+    with open(pipeline_spec_file) as f:
+        spec = json.load(f)
+    for component in [spec['root']] + list(spec['components'].values()):
         if 'dag' in component:
             for task in component['dag']['tasks'].values():
                 task['cachingOptions'] = {'enableCache': enable_caching}
+    job = {
+        'pipelineSpec': spec,
+        'runtimeConfig': {
+            'parameterValues': {},
+        },
+    }
```

Comment:
Here we missed a case to fill in runtimeConfig.parameterValues using default value for pipeline inputs.